### ISSUE TITLE
Deployment fails with lock file in custom location

### ISSUE THREAD

**Issue by chlebowa** [rstudio/rsconnect#1122]
I am running a `shiny` app in an R project let's call it the parent app. The project has a `renv.lock` file but it is not located in the project root. The environment variable `RENV_PATHS_LOCKFILE` is set in the `.Rprofile` of the project, before `renv/activate.R` is run. So far, so good.

The parent app generates code for another `shiny` app, let's call this one the daughter app. The daughter app file is placed in a separate directory in within the project. From there the daughter app is deployed.

This process fails because the lock file is not found. It does work when the lock file is copied from the location specified by the env var to the project root. Note that there is no attempt at creating a lock file - it is not found where expected and an error is raised.

Can this be fixed so that custom location of the lock file is taken into account?

**Comment by LDSamson** [comment]
Just a note that the same issue occurs if `renv` profiles are used, because then the `renv` lockfile is not in the root folder but in the `renv/profiles/name_of_profile` folder

**Comment by toph-allen** [comment]
Hey @chlebowa. Just to confirm my understanding, it sounds like the core of your question is to do with the location of the lockfile, and the actual mechanisms (dynamic creation of a child app) are secondary? (The fact that this also happens when `renv` profiles are in use makes me think so.)

If you're able to easily provide a minimal reproducible example that'd be helpful.

**Comment by chlebowa** [comment]
Correct, the problem is the location of the file. The rest is the full context.

I've prepared a [toy package](https://github.com/chlebowa/reprex) to reproduce the issue.

<details><summary> deployment </summary>

```
rsconnect::deployApp(
  appDir = system.file("shiny", package = "reprex"),
  appName = "reprex",
  appTitle = "examlpe shiny app",
  account = "<ACCOUNT>",
  launch.browser = FALSE,
  lint = FALSE,
  forceUpdate = TRUE
)
```

</details>

```
── Preparing for deployment ─────────────────────────────────────────────────────────────────────────────────
✔ Deploying "reprex" using "server: shinyapps.io / username: ***"
ℹ Creating application on server...
✔ Created application with id 14485996
ℹ Bundling 1 file: app.R
ℹ Capturing R dependencies
Error in open.connection(con, "rb") : cannot open the connection
In addition: Warning message:
In open.connection(con, "rb") :
  cannot open file '/tmp/RtmpQ3dp4D/file3a1f25e162f23/renv.lock': No such file or directory
```

**Comment by toph-allen** [comment]
Thanks, I'll take a look at it.

**Comment by toph-allen** [comment]
Some notes from looking at this with @jonkeane:

- When finding dependencies, we get the lockfile location here: https://github.com/rstudio/rsconnect/blob/2ac4925a96520254551797adc64e3762a3a1ab1a/R/bundlePackage.R#L48-L71
- The function that gets the lockfile (https://github.com/rstudio/rsconnect/blob/2ac4925a96520254551797adc64e3762a3a1ab1a/R/bundlePackageRenv.R#L151-L153) should respect the `RENV_PATHS_LOCKFILE` environment var.

**Comment by chlebowa** [comment]
This looks right. Would you like me to prepare a patch or do you want to take care of it yourselves?

**Comment by jonkeane** [comment]
If you're interested and willing, that would be lovely. We're here for review + help if you need it

**Comment by chlebowa** [comment]
All right 👍

**Comment by toph-allen** [comment]
Thanks! As @jonkeane said, we're available if you need help.
